src/Monolog/Logger.php

```diff
@@ -285,9 +285,11 @@ public function addRecord($level, $message, array $context = array())
         foreach ($this->processors as $processor) {
             $record = call_user_func($processor, $record);
         }
-        while (isset($this->handlers[$handlerKey]) &&
-            false === $this->handlers[$handlerKey]->handle($record)) {
-            $handlerKey++;
+
```

Comment:
this change kills the optimization which line https://github.com/Seldaek/monolog/pull/641/files#diff-dce565f403e044caa5e6a0d988339430R258 tries to achieve